### PR TITLE
Add support for entropy

### PR DESCRIPTION
**Pull request by 42wim**
This adds `xxxEntropy` methods that allow an entropy parameter.
Not sure about the method naming :-)

### CODE REVIEW

File: dpapi.go [hunk 1]
```diff
@@ -61,12 +65,25 @@ func encrypt(secret string, cf cryptProtect) (string, error) {
 
 // EncryptBytes encrypts a byte array and returns a byte array
 func EncryptBytes(data []byte) ([]byte, error) {
-	return encryptBytes(data, cryptProtectUIForbidden)
+	return encryptBytes(data, []byte{}, cryptProtectUIForbidden)
```
Comment (dcormier) [inline dpapi.go]:
```suggestion
	return encryptBytes(data, nil, cryptProtectUIForbidden)
```

No need to allocate an empty `[]byte` here. Checking the `len` of a `nil` `[]byte` is also 0.

File: dpapi.go [hunk 2]
```diff
@@ -77,19 +94,35 @@ func encryptBytes(data []byte, cf cryptProtect) ([]byte, error) {
 // EncryptBytesMachineLocal encrypts a byte array and returns a byte array and associates the data
 // encrypted with the current computer instead of with an individual user.
 func EncryptBytesMachineLocal(data []byte) ([]byte, error) {
-	return encryptBytes(data, cryptProtectUIForbidden|cryptProtectLocalMachine)
+	return encryptBytes(data, []byte{}, cryptProtectUIForbidden|cryptProtectLocalMachine)
```
Comment (dcormier) [inline dpapi.go]:
```suggestion
	return encryptBytes(data, nil, cryptProtectUIForbidden|cryptProtectLocalMachine)
```

File: dpapi.go [hunk 3]
```diff
@@ -99,12 +132,25 @@ func DecryptBytes(data []byte) ([]byte, error) {
 
 // Decrypt a string to a string
 func Decrypt(data string) (string, error) {
+	return DecryptEntropy(data, "")
+}
+
+// EncryptBytes encrypts a byte array and returns a byte array
+func DecryptBytes(data []byte) ([]byte, error) {
+	return decryptBytes(data, []byte{}, cryptProtectUIForbidden)
```
Comment (dcormier) [inline dpapi.go]:
```suggestion
	return decryptBytes(data, nil, cryptProtectUIForbidden)
```

**Comment by billgraziano** [comment]
Thank you!!

**Pull request merged**